File: 백준/Silver/10799. 쇠막대기/쇠막대기.js
const findLaser = (input) => { 
  let stack = [];
  let count = 0;

  for (let i = 0; i < input.length; i++) {
    const char = input[i];
    
    if (char === "(") {
      stack.push(i); 
    } else if (char === ")") {
      if (stack.length > 0) {
        const lastOpenIdx = stack.pop(); 
        if (i - lastOpenIdx === 1) {
          count += stack.length; 
        } else {
          
          count++; 
        }
      }
    }
  }

  console.log(count);
}
const readline = require('readline')
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

let input

rl.on("line", (line) => { 
    input = [...line]
}).on("close", () => { 
    findLaser(input)
    process.exit()
})

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
def soluction(arr):
    stack = []
    cnt =0
    idx =0
    for idx in range(len(arr)):
        
        if arr[idx] == '(':
            stack.append('(')
        else:
            if arr[idx-1]=='(':
                stack.pop()
                cnt += len(stack)
            else:
                stack.pop()
                cnt+=1
        
    print(cnt)

arr = list(input())
soluction(arr)